core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -2852,18 +2853,23 @@ struct ViewValueFunctor<ExecSpace, ValueType, false /* is_scalar */> {
   ViewValueFunctor& operator=(const ViewValueFunctor&) = default;
 
   ViewValueFunctor(ExecSpace const& arg_space, ValueType* const arg_ptr,
-                   size_t const arg_n)
-      : space(arg_space), ptr(arg_ptr), n(arg_n), destroy(false) {}
+                   size_t const arg_n, std::string arg_name)
+      : space(arg_space),
+        ptr(arg_ptr),
+        n(arg_n),
+        destroy(false),
+        name(std::move(arg_name)) {}
 
   void execute(bool arg) {
     destroy = arg;
     if (!space.in_parallel()) {
+      std::string functor_name = (destroy ? "Kokkos::View::destruction"
+                                          : "Kokkos::View::initialization");
```

Comment:
A bit of a nitpick: I don't like it when people put different types in the second and third positions of the ternary operator. Use `auto` on the LHS and construct the `std::string` inside of the ternary operator.

Reply:
> I don't like it when people put different types in the second and third positions of the ternary operator

But that wouldn't compile would it?

Reply:
It will if they both have implicit decay to the same type; here it's `const char[N]` and `const char[M]` decaying to `const char*`.

Reply:
Ohhhh, that's fascinating, I didn't pick up on that. Would explicitly constructing a string on each side of the ternary be more comfortable?